### PR TITLE
feat: implemented get supported regions feature

### PR DESCRIPTION
**Pull request by imacpowers**
### What does this PR do?

This PR implements an endpoint for getting the list of supported regions:

- **Added GET /api/v1/regions Endpoint:** Returns the list of supported regions.
- **Response Handling:** Provides appropriate responses for missing or invalid refresh tokens.
- **Error Handling:** Returns specific HTTP status codes and messages based on authorization status.

### How should this be manually tested?

1. **Register a User:**
   - Use the `POST /api/v1/auth/register` endpoint to register a new user.

2. **Login User:**
   - Make a `POST` request to `/api/v1/auth/login` with the registered user's password and email in the request body.
   - Check the response for appropriate status codes and messages.

3. **Get Supported Regions:**
   - Make a `GET` request to `/api/v1/regions` with a valid token.
   - Verify the response contains the list of supported regions.

4. **Edge Cases:**
   - **Invalid Token:** Test with an invalid token to ensure the endpoint returns a `401 Unauthorized` status with the correct message.
   - **Missing Token:** Test without a token to ensure the endpoint returns a `401 Unauthorized` status with the correct message.

### Checklist

- [x] Implemented `GET /api/v1/regions` endpoint in the `regions.controller.ts`.
- [x] Added `findAll` method in the `RegionsService`.
- [x] Modified `regions.controller.ts` to provide appropriate responses for various cases.
- [x] Tested endpoint functionality to ensure it correctly handles valid and invalid tokens.
- [x] Added appropriate HTTP status codes and response messages for edge cases.
- [x] Added tests for this endpoint.

### Response Examples

**Success:**
```json
{
  "status": "success",
  "data": {
    "regions": [
      {
        "id": "string",
        "region": "string"
      }
    ]
  }
}
```

**Error:**
```json
{
  "status": "error",
  "message": "You are not authorized for this action",
  "status_code": 401
}
```

### Link to Issue

Link to this issue: #34

### CODE REVIEW

File: src/modules/regions/regions.service.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import { Injectable } from '@nestjs/common';
```
Comment (Cobidex) [inline src/modules/regions/regions.service.ts:1]: Please stick to the agreed field name for status codes which is status_code

Comment (imacpowers) [reply src/modules/regions/regions.service.ts:1]: i've fixed that

**Review by davidolanrewaju — APPROVED** [review]

**Review by TheCodeGhinux — APPROVED** [review]
Doesn't look good to me. lots of unnecessary changes

**Review by TheCodeGhinux — COMMENTED** [review]
lots of unnecessary changes

**Comment by imacpowers** [comment]
@TheCodeGhinux , most of the unnecessary changes were attempts at fixing merge conflicts.
